packages/superset-ui-core/test/query/processExtraFormData.test.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  appendExtraFormData,
+  overrideExtraFormData,
+} from '@superset-ui/core/src/query/processExtraFormData';
+
+describe('appendExtraFormData', () => {
+  it('should add allowed values to non-existent value', () => {
+    expect(
+      appendExtraFormData(
+        {
+          datasource: 'table_1',
+          granularity: 'something',
+          viz_type: 'custom',
+        },
+        {
+          filters: [{ col: 'my_col', op: '==', val: 'my value' }],
+        },
+      ),
+    ).toEqual({
+      datasource: 'table_1',
+      filters: [{ col: 'my_col', op: '==', val: 'my value' }],
+      granularity: 'something',
+      viz_type: 'custom',
+    });
+  });
+
+  it('should add allowed values to pre-existent value', () => {
```

Comment:
```suggestion
  it('should add allowed values to preexisting value', () => {
```
...not that you were looking for grammar nits...